packages/front-end/components/Experiment/CustomFieldDisplay.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { FC, useState } from "react";
+import {
+  CustomExperimentField,
+  ExperimentInterfaceStringDates,
+} from "back-end/types/experiment";
+import { useForm } from "react-hook-form";
+import { CustomField } from "back-end/types/organization";
+import { useUser } from "@/services/UserContext";
+import { useCustomFields } from "@/services/experiments";
+import { useAuth } from "@/services/auth";
+import PremiumTooltip from "@/components/Marketing/PremiumTooltip";
+import HeaderWithEdit from "../Layout/HeaderWithEdit";
+import Modal from "../Modal";
+import CustomFieldInput from "./CustomFieldInput";
+
+const CustomFieldDisplay: FC<{
+  label?: string;
+  canEdit?: boolean;
+  mutate?: () => void;
+  experiment: ExperimentInterfaceStringDates;
+}> = ({ label = "Additional Fields", experiment, canEdit = true, mutate }) => {
+  const [editModal, setEditModal] = useState(false);
+  const customFields = useCustomFields();
+  const customFieldsMap = new Map();
+  const defaultFields: CustomExperimentField = {};
+  if (customFields && customFields.length) {
+    customFields.map((v) => {
+      defaultFields[v.id] =
+        v.type === "boolean"
+          ? JSON.stringify("")
+          : v.type === "multiselect"
+          ? JSON.stringify([])
+          : "";
+      customFieldsMap.set(v.id, v);
+    });
+  }
+
+  const { hasCommercialFeature } = useUser();
+  const hasCustomFieldAccess = hasCommercialFeature("custom-exp-metadata");
+  const form = useForm<Partial<ExperimentInterfaceStringDates>>({
+    defaultValues: {
+      customFields: experiment?.customFields || defaultFields,
+    },
+  });
+  const { apiCall } = useAuth();
+
+  if (customFields && customFields.length) {
+    return (
+      <>
+        {editModal && (
+          <Modal
+            header={"Edit Custom Fields"}
```

Comment:
For some reason this modal renders empty for me:
<img width="927" alt="image" src="https://user-images.githubusercontent.com/7927873/219818426-0426b58d-4aee-4fc9-9ad3-d1c2b53995b2.png">

Reply:
I think your filter function in `CustomFieldInput.tsx` is skipping the "all projects" state (no project set).

Reply:
Also, might want a null state in the UI

Reply:
I added a case for the null state in the UI. The question is if a custom field has a project assigned, and the experiment is 'all projects', should that field show up?

Reply:
1. Still not seeing the null UI, just an empty modal.
2. When modifying the current experiment to have a project (and there is a custom field matching that project), I still get the empty modal above.
3. My IDE is showing some warnings about `form.watch("project")` and `customFields.${v.id}`. Actually `selectedProject` is always `undefined` for me regardless of the experiment's set project (or nav project selector).

for yoru question:
- If the custom field has a project and the experiment is "all projects", I'd probably not show the field.
- If the custom field is all projects and the experiment has a project, I'd probably show it.

Reply:
You can fix the undefined project by modifying the form default values to include
```
      project: experiment?.project || "",
```
(CustomFieldDisplay.tsx and EditInfoForm.tsx).

Just need to make the filtering more lenient for fields with no projects.

Reply:
Thanks for the comments. I believe I've fixed the feedback in the latest commit.